jcl/src/jdk.management/share/classes/openj9/lang/management/internal/OpenJ9DiagnosticsMXBeanImpl.java

```diff
@@ -0,0 +1,422 @@
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+/*[ELSE]
+		case "java": //$NON-NLS-1$
+			try {
+				fileName = Dump.javaDumpToFile(fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+		case "heap": //$NON-NLS-1$
+			try {
+				fileName = Dump.heapDumpToFile(fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+		case "system": //$NON-NLS-1$
+			try {
+				fileName = Dump.systemDumpToFile(fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+		case "snap": //$NON-NLS-1$
+			try {
+				fileName = Dump.snapDumpToFile(fileNamePattern);
+			} catch (Exception e) {
+				handleInvalidDumpOptionException(e);
+				throw handleError(e);
+			}
+			break;
+/*[ENDIF]*/
+		default:
+			/*[MSG "K0663", "Invalid or Unsupported Dump Agent cannot be triggered"]*/
+			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0663")); //$NON-NLS-1$
+		}
+		return fileName;
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String triggerClassicHeapDump() throws InvalidOptionException {
+		String dumpOptions = "heap:opts=CLASSIC"; //$NON-NLS-1$
+		checkManagementSecurityPermission();
+		try {
+/*[IF Sidecar19-SE]*/
+			String fileName = (String)dump_triggerDump.invoke(null, dumpOptions);
+/*[ELSE]
+			String fileName = Dump.triggerDump(dumpOptions);
+/*[ENDIF]*/
+			return fileName;
+		} catch (Exception e) {
+
+			handleInvalidDumpOptionException(e);
+			throw handleError(e);
+		}
+	}
+
+	private static void checkManagementSecurityPermission() {
+		/* Check the caller has Management Permission. */
+		SecurityManager manager = System.getSecurityManager();
+		if( manager != null ) {
+/*[IF Sidecar19-SE]*/
+			try {
+				Object controlPermission = managementPermission.get(null);
+				manager.checkPermission((java.security.Permission)controlPermission);
+			} catch (Exception e) {
+				throw handleError(e);
+			}
+/*[ELSE]
+			manager.checkPermission(ManagementPermissionHelper.MPCONTROL);
+/*[ENDIF]*/
+		}
+	}
+
+	/**
+	 * Singleton accessor method. Returns an instance of {@link OpenJ9DiagnosticsMXBeanImpl}.
+	 *
+	 * @return a static instance of {@link OpenJ9DiagnosticsMXBeanImpl}.
+	 */
+	public static OpenJ9DiagnosticsMXBeanImpl getInstance() {
+		return instance;
+	}
+
+	/**
+	 * Empty constructor intentionally private to prevent instantiation by others.
+	 */
+	private OpenJ9DiagnosticsMXBeanImpl() {
+		super();
+	}
+
+	/**
+	 * Returns the object name of the MXBean.
+	 *
+	 * @return objectName representing the MXBean.
+	 */
+	@Override
+	public ObjectName getObjectName() {
+		try {
+			ObjectName name = new ObjectName("openj9.lang.management:type=OpenJ9Diagnostics");  //$NON-NLS-1$
+			return name;
+		} catch (MalformedObjectNameException e) {
+			return null;
```

Comment:
This should probably throw an Error rather than return null.

Reply:
Other MXBeans such as JVMCpuMonitor, ProcessorMXBeanImpl all have similar logic and return null on MalformedObjectNameException. Hence used the same logic here. Should I change it to throw an error?

Reply:
I think they all should throw an Error, but I'm fine leaving that cleanup for later.

Reply:
Thanks Keith. I have made the other changes as per your review comments. Please review again.